lib/matplotlib/dates.py

```diff
@@ -1241,10 +1241,7 @@ def __init__(self, bymonthday=None, interval=1, tz=None):
 
         Default is to tick every day of the month: ``bymonthday=range(1,32)``
         """
-        if interval < 1:
-            raise ValueError("The interval parameter must be an integer "
-                             "greater than or equal to one.")
-        if not isinstance(interval, int):
+        if interval < 1 or not isinstance(interval, int):
             raise ValueError("The interval parameter must be an integer "
                              "greater than or equal to one.")
```

Comment:
Basically if for whatever reason interval was passed in as a string, like it was 'a', the comparison
 'a'<=1 would throw a 'TypeError'. By checking that it's an integer first, that part of the conditional will fail and therefore the second check (interval<=1) won't ever happen.